pkgs/applications/editors/rstudio/use-system-node.patch

```diff
@@ -0,0 +1,28 @@
+--- a/src/gwt/build.xml
++++ b/src/gwt/build.xml
+@@ -84,23 +84,7 @@
+    </target>
+
+    <!-- panmirror typescript library -->
+-   <!-- ensure version matches RSTUDIO_NODE_VERSION -->
+-   <property name="node.version" value="14.17.5"/>
+-   <property name="node.dir" value="../../dependencies/common/node/${node.version}"/>
+-   <condition property="node.bin" value="../../../${node.dir}/bin/node">
+-     <not>
+-        <os family="windows" />
+-     </not>
+-   </condition>
+-   <condition property="node.bin" value="${node.dir}/node.exe">
+-     <os family="windows" />
+-   </condition>
+-
+-   <!-- use node from /opt/rstudio-tools if installed (typical for Docker) -->
+-   <available
+-      property="node.bin"
+-      value="/opt/rstudio-tools/dependencies/common/node/${node.version}/bin/node"
+-      file="/opt/rstudio-tools/dependencies/common/node/${node.version}/bin/node"/>
++   <property name="node.bin" value="/usr/bin/node"/>
```

Comment:
I'm not sure if this is appropriate. Can we guarantee that `/usr/bin/node` exists after installing this?

Reply:
Well, I'm patching the "build.xml" so in my understanding node is only mandatory during build time, and the java script or its derived sub-languages are executed through the qtwebengine.

Reply:
The current system does not contain nodejs as a runtime [dependency_graph.txt](https://github.com/NixOS/nixpkgs/files/8889155/dependency_graph.txt).

Reply:
During build time nodejs is available within setting in `nativeBuildInputs`: [default.nix](https://github.com/NixOS/nixpkgs/blob/bbfc6883de78c44c1d59221668e4e6b396377389/pkgs/applications/editors/rstudio/default.nix)

> 
    nativeBuildInputs = [
      cmake
      unzip
      ant
      jdk
      makeWrapper
      pandoc
      nodejs
    ] ++ lib.optional (!server) [
      copyDesktopItems
    ];

Reply:
Still I don't see why not supplying node as `${pkgs.node}/bin/node` or similar

Reply:
It is not clear, why you want to change this variable. We are not in the scope of `pkgs` and working with `nodejs` and not `node`.

Reply:
Sorry, yes, `pkgs.nodejs`. I thought that it is possible to interpolate in a patch file. Maybe I'm wrong, in that case we can still do `substituteInPlace` afterwards I think.

Reply:
Is it not a classic problem of Nix working with absolute paths and FHS? I would not change the behaviour, because it is very "understandable", and show how to handle the dynamic aspect of Nix building process. It is not hidden inside a patch.

Reply:
> I would not change the behaviour, because it is very "understandable", a

@MrTarantoga What behaviour do you not want to change? 

> Maybe I'm wrong, in that case we can still do substituteInPlace afterwards I think.

I don't see a problem using `substitutInPlace` as it is done with `pandoc` https://github.com/NixOS/nixpkgs/pull/177021/files#diff-7a68076d70e2a1aaeee753bf78956eb8e77a550121f5fca11bc700a28f88644eR146

> my understanding node is only mandatory during build time,

@MrTarantoga Maybe I got you wrong, but even if it is only mandatory during build time, `nativeBuildInputs` are not available in `/usr/bin` even during the build phase.

Reply:
Available in `PATH` no in `/usr/bin/`.

Reply:
Yes, and the nix path is replaced [here](https://github.com/NixOS/nixpkgs/blob/bbfc6883de78c44c1d59221668e4e6b396377389/pkgs/applications/editors/rstudio/default.nix#L139)

Reply:
I would suggest doing something like we already do for clang and R, i.e in the patch file change the path to `@node@` and then in the `substituteInPlace` change this to the nix store path. It's more explicit than the current approach.

Reply:
@MrTarantoga sorry I missed that. In that case I don't understand @turion point.

Reply:
Ah, I hadn't seen that part! Then everything is fine. Sorry for the confusion.

Reply:
I create a patch for this.

Reply:
Fixed in https://github.com/NixOS/nixpkgs/pull/177021/commits/f4c5c86fbbe264d67e4eb416add0a7eee17d3f49